src/functionaltest/java/com/ericsson/ei/frontend/pageobjects/SubscriptionPage.java

```diff
@@ -140,28 +148,53 @@ public boolean presenceOfClickGetTemplateButton() {
-                request -> (request).getURI().toString().contains("/subscriptionsTemplate")));
-        new WebDriverWait(driver, TIMEOUT_TIMER).until(ExpectedConditions.elementToBeClickable(By.className("get_subscription_template")));
+        Mockito.doReturn(response).when(mockedHttpClient)
+                .execute(Mockito.argThat(request -> (request).getURI().toString().contains("/subscriptionsTemplate")));
+        new WebDriverWait(driver, TIMEOUT_TIMER)
+                .until(ExpectedConditions.elementToBeClickable(By.className("get_subscription_template")));
         WebElement getTemplateButton = driver.findElement(By.className("get_subscription_template"));
         getTemplateButton.click();
     }
 
     public void clickViewBtn() {
-        new WebDriverWait(driver, TIMEOUT_TIMER).until(ExpectedConditions.elementToBeClickable(By.xpath("//button[contains(text(),'View')]")));
+        new WebDriverWait(driver, TIMEOUT_TIMER)
+                .until(ExpectedConditions.elementToBeClickable(By.xpath("//button[contains(text(),'View')]")));
         WebElement viewBtn = driver.findElement(By.xpath("//button[contains(text(),'View')]"));
         viewBtn.click();
     }
 
     public void clickFormCloseBtn() {
+
         new WebDriverWait(driver, TIMEOUT_TIMER).until(ExpectedConditions.elementToBeClickable(By.className("close")));
         WebElement viewBtn = driver.findElement(By.className("close"));
         viewBtn.click();
     }
 
     public String getSubscriptionNameFromSubscription() {
-        new WebDriverWait(driver, TIMEOUT_TIMER).until(ExpectedConditions.elementToBeClickable(By.xpath("//tr[@class='odd']/td[2]")));
+        new WebDriverWait(driver, TIMEOUT_TIMER)
+                .until(ExpectedConditions.elementToBeClickable(By.xpath("//tr[@class='odd']/td[2]")));
         WebElement subscriptionNameElement = driver.findElement(By.xpath("//tr[@class='odd']/td[2]"));
         return subscriptionNameElement.getText();
     }
+
+    public Boolean buttonExist(String loc) {
+        try {
+            wait.until(ExpectedConditions.elementToBeClickable(By.xpath(loc)));
+        } catch (Exception e) {
+            return false;
+        }
+        return true;
+    }
+
+    public void clickReloadLDAP(String response, String responseAuth) throws IOException {
+        CloseableHttpResponse responseData = this.createMockedHTTPResponse(response, 200);
+        CloseableHttpResponse responseDataAuth = this.createMockedHTTPResponse(responseAuth, 200);
+        Mockito.doReturn(responseData).when(mockedHttpClient).execute(
+                Mockito.argThat(request -> ((HttpRequestBase) request).getURI().toString().contains("subscriptions")));
+        Mockito.doReturn(responseDataAuth).when(mockedHttpClient)
+                .execute(Mockito.argThat(request -> ((HttpRequestBase) request).getURI().toString().contains("auth")));
+        WebElement reloadBtn = wait.until(ExpectedConditions.elementToBeClickable(By.id("reloadButton")));
```

Comment:
Dont use wait.until with an already initialized WebdriverWait. It doesnt reset between each use. You have to initialize a new for every timer you create. Use a "new WebDriverWait" instead.

Reply:
OK